### PR TITLE
Implementing basic FeedCategory model and tests

### PR DESCRIPTION
**Pull request by zach-delong**
This history (and diff) is much cleaner now.  I have a hunch of what happened to the last one, so I'll avoid that moving forward.

### CODE REVIEW

File: models/index.js [hunk 1]
```diff
@@ -6,12 +6,15 @@ var path = require('path');
...
+  models.FeedCategory = sequelize.import(path.join(__dirname, 'FeedCategory.js'));
   models.User      = sequelize.import(path.join(__dirname, 'User.js'));
   models.Feed      = sequelize.import(path.join(__dirname, 'Feed.js'));
   models.AccessKey.sync();
+  models.FeedCategory.sync();
   models.User.sync();
   models.Feed.sync();
   models.Feed.belongsTo(models.User, { foreignKey: 'owner_uid' });
   models.AccessKey.belongsTo(models.User, { foreignKey: 'owner_uid' });
+  models.FeedCategory.belongsTo(models.User, { foreignKey: 'owner_uid' });
```
Comment (jchristi) [inline models/index.js]: Missing association for ParentCategory (parent_cat)

**Comment by jchristi** [comment]
Add association for parent category and a relevant test case(s)

**Comment by ZacheryPD** [comment]
I went ahead and implemented the missing association, good catch!  Will this work, or do we also need an association to represent the other side of the relationship?  It looks like Sequelize can do both but I'm not familiar enough to say if we want to do that or not.

**Comment by jchristi** [comment]
Whereas a category can only have one parent, a category can have many children. I think both associations should be defined.

**Comment by ZacheryPD** [comment]
@jchristi this should be resolved now.  Make sure you take a look at the names of the associations, did I do that right?  I think I did, but I expected that syntax to be different...

File: tests/models/FeedCategory_test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    expect(key.id).to.equal(1);
+  }));
+
+  describe('associations', function() {
+    it('has User property', co.wrap(function* (){
+      let sequelize = yield getTestDB();
+      expect(sequelize.models.FeedCategory.associations).to.have.property('User');
+    }));
+
+    it('has a FedCategory (parent) property', co.wrap(function* (){
```
Comment (jchristi) [inline tests/models/FeedCategory_test.js]: keep the test title simple: `has Parent property`

File: tests/models/FeedCategory_test.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      let sequelize = yield getTestDB();
+      expect(sequelize.models.FeedCategory.associations).to.have.property('Parent');
+    }));
+
+    it('has a FedCategory (children) property', co.wrap(function* (){
```
Comment (jchristi) [inline tests/models/FeedCategory_test.js]: `has Children property`

**Review by jchristi — CHANGES_REQUESTED** [review]
minor text changes

**Comment by jchristi** [comment]
@ZacheryPD the association names can be whatever we want. I'm debating whether they should be all lowercased or not.

**Review by jchristi — APPROVED** [review]

**Pull request merged**